### ISSUE TITLE
Unexpected behavior of buffer with LinearRing object

### ISSUE THREAD

**Issue by pohlt** [shapely/shapely#790]
## Expected behavior and actual behavior.

Maybe I'm using the LinearRing in an unsupported fashion, but when I use `LinearRing.buffer` I get more vertices than expected around the first (identical to the last since it's a ring) input vertex.

## Steps to reproduce the problem.

``` python
# define an equilateral triangle as LinearRing (just to make sure than all three corners are comparable
lr = shapely.geometry.LinearRing([[0, 0],
                                  [50, (100**2-50**2)**0.5],
                                  [100, 0]])
print(lr.is_closed)  # is closed as expected
print(list(lr.coords))  # the three given coordinates plus the closing [0, 0]
lr_buffered = lr.buffer(2, 2)
print(list(lr_buffered.exterior.coords))  # UNEXPECTED: more vertices than expected around [0, 0]
```

The output for the last `print` is a mess, but let me describe it here:

1. 4 vertices around the first input vertex.
2. 4 vertices around the second input vertex.
3. 4 vertices around the third input vertex.
4. **Another 8 vertices around the fourth/first input vertex.**

## Operating system

Linux manjaro 5.3.7-2-MANJARO SMP PREEMPT Tue Oct 22 18:56:05 UTC 2019 x86_64 GNU/Linux

## Shapely version and provenance

1.6.4.post2 installed from PyPI using pip

*Thanks for your work on shapely and for looking into this issue.*

**Comment by mwtoews** [comment]
I should note that this behavior is found with JTS too. Here is what I get with JTS test builder, with a buffer distance of 10:
![buff](https://i.imgur.com/8hKaOY2.png)
The extra points are somewhat explained with the buffer curve operation (i.e. `parallel_offset`), which shows some interesting patterns around the closed location:
![buffcurve](https://i.imgur.com/zfttmto.png)
These extra points can be avoided by `resolution` to a multiple of three, i.e. 3, 6, 9, 12, 15. Any other value will add the extra points.

**Comment by sgillies** [comment]
@mwtoews thanks for digging into this! Do you think we should consider constraining the resolution?

**Comment by mwtoews** [comment]
@sgillies users should be able to choose any `resolution` they like. This particular geometry only works with a multiple of three due to the 60° angle of incidence at the closed location. If this were a exterior of a rectangle, any `resolution` value looks good, without adding extra points. But if this were any other geometry with an imperfect angle of incidence at the closed location that is not divisible by 180°, then most `resolution` values would add extra points near the closed location.

Probably the best way to avoid these extra points is to buffer a polygon, if possible, as it doesn't seem to suffer the same limitation.

**Comment by dr-jts** [comment]
As it happens I just encountered this issue while working on the new JTS [VariableBuffer](https://github.com/locationtech/jts/blob/master/modules/core/src/main/java/org/locationtech/jts/operation/buffer/VariableBuffer.java) implementation.  I solved the problem there by always generating the arc points using the same locations around the circle.  

However, a more appropriate fix in this case might be to have the buffer code detect a LinearRing and generate a closed curve for it (as is done for the rings of polygons).
